test/conformance/tests/test_exporter_metrics.go

```diff
@@ -0,0 +1,183 @@
+
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/cluster"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/discovery"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/namespaces"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/network"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/pod"
+
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("[sriov] Metrics Exporter", Ordered, func() {
+
+	BeforeAll(func() {
+		if cluster.VirtualCluster() {
+			Skip("IGB driver does not support VF statistics")
+		}
+
+		err := namespaces.Create(namespaces.Test, clients)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = namespaces.Clean(operatorNamespace, namespaces.Test, clients, discovery.Enabled())
+		Expect(err).ToNot(HaveOccurred())
+
+		featureFlagInitialValue := isFeatureFlagEnabled("metricsExporter")
+		DeferCleanup(func() {
+			By("Restoring initial feature flag value")
+			setFeatureFlag("metricsExporter", featureFlagInitialValue)
+		})
+
+		By("Enabling `metricsExporter` feature flag")
+		setFeatureFlag("metricsExporter", true)
+
+		WaitForSRIOVStable()
+	})
+
+	It("collects metrics regarding receiving traffic via VF", func() {
+		sriovInfos, err := cluster.DiscoverSriov(clients, operatorNamespace)
+		Expect(err).ToNot(HaveOccurred())
+
+		node, nic, err := sriovInfos.FindOneSriovNodeAndDevice()
+		Expect(err).ToNot(HaveOccurred())
+		By("Using device " + nic.Name + " on node " + node)
+
+		_, err = network.CreateSriovPolicy(clients, "test-me-policy-", operatorNamespace, nic.Name, node, 2, "metricsResource", "netdevice")
+		Expect(err).ToNot(HaveOccurred())
+
+		err = network.CreateSriovNetwork(clients, nic, "test-me-network", namespaces.Test, operatorNamespace, "metricsResource", ipamIpv4)
+		Expect(err).ToNot(HaveOccurred())
+		waitForNetAttachDef("test-me-network", namespaces.Test)
+
+		pod := createTestPod(node, []string{"test-me-network"})
+
+		ips, err := network.GetSriovNicIPs(pod, "net1")
+		Expect(err).ToNot(HaveOccurred())
+		Expect(ips).NotTo(BeNil(), "No sriov network interface found.")
+		Expect(len(ips)).Should(Equal(1))
+
+		initialMetrics := getMetricsForNode(node)
+		initialRxBytes := getCounterForPod(initialMetrics, pod, "sriov_vf_rx_bytes")
+		initialRxPackets := getCounterForPod(initialMetrics, pod, "sriov_vf_rx_packets")
+
+		for _, ip := range ips {
+			pingPod(ip, node, "test-me-network")
+		}
+
+		finalMetrics := getMetricsForNode(node)
+		finalRxBytes := getCounterForPod(finalMetrics, pod, "sriov_vf_rx_bytes")
+		finalRxPackets := getCounterForPod(finalMetrics, pod, "sriov_vf_rx_packets")
+
+		Expect(finalRxBytes).Should(BeNumerically(">", initialRxBytes))
+		Expect(finalRxPackets).Should(BeNumerically(">", initialRxPackets+3))
```

Comment:
Because `pingPod(...)` function sends 3 ICMP packets

https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/a8476f0c98217b39f6c1adaf79c11efc288d36bb/test/conformance/tests/test_sriov_operator.go#L2652-L2669

I could refactor that function a bit to make this information more explicit

Reply:
is it not enough to keep the check > initialRxPackets ?

Reply:
yes, it is. It would also be clearer as the number "3" is hidden in another function.

fixing